### PR TITLE
Implements Loading State for Remaining Pages

### PR DESCRIPTION
**Pull request by Monica-Coira**
## Description

- Implements loading state for the Calendar, Login, and Signup pages.
- Displays a loading spinner when a user clicks the Signup, Login, or Submit Availability buttons.
- In the next PR, I will continue working on deploying my app so that users can access it on their devices.

## Milestones
This works towards implementing a loading state throughout the app.

## Test Plan

https://github.com/user-attachments/assets/5650fcfb-f08a-485c-ad48-8a3d088b63fd

These are the corner cases I tested:

- Ensured that the loading spinner is displayed when users click a button and immediately removed from the screen when the data is successfully fetched and displayed.
- Ensured that a loading spinner is displayed every time data is being fetched or is not able to be displayed yet.
- Ensured that the loading spinner is visible to the user on every page when it is displayed.

### CODE REVIEW

File: client/src/components/Calendar.jsx [hunk 1]
```diff
@@ -146,13 +148,15 @@ const Calendar = () => {
     }, [user])
 
     const matchByAvailability = async () => {
+        setIsLoading(true);
         await MatchByAvailability(fetchData, allUsers);
         CompatibilityScore(fetchData, user);
+        setIsLoading(false);
```
Comment (truongh) [inline client/src/components/Calendar.jsx:154]: If there is an error with `MatchByAvailability` or `CompatibilityScore` , the loading could run indefinitely, better we wrap it in a  `try ... catch` block

Comment (Monica-Coira) [reply client/src/components/Calendar.jsx:154]: Got it, just implemented this change in PR #56 .
